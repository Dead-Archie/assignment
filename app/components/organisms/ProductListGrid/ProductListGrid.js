import React from 'react';
import PropTypes from 'prop-types';
import { ImageVanilla as Image } from '@xt-pagesource/atomic-react-pattern-lib';
import withStyles from '../../../lib/withStyles';
import styles from './ProductListGrid.style';

class ProductListGrid extends React.Component {
  renderSections = (key, values) => {
    return (
      <div className="product-row">
        <div className="product-key">{key}</div>
        <div className="product-value">{values}</div>
      </div>
    );
  };

  render() {
    const { product, className, queryOrigin } = this.props;
    const { id, name, status, species, gender, origin, location, image } = product;
    const productArry = [
      { Status: status },
      { Species: species },
      { Gender: gender },
      { Origin: origin.name || '' },
      { 'Last Location': location.name || '' },
    ];

    if (queryOrigin && queryOrigin !== origin.name) {
      return false;
    }
    return (
      <div className={`${className} col-6-small`}>
        <div className="container-Background">
          <div className="image-container">
            <Image src={image} className="product-image" />
            <div className="product-name">
              <div className="product-title">{name}</div>
              <span>{`ID :${id}`}</span>
            </div>
          </div>
          <div className="info-container">
            {productArry.map(item => {
              return (
                <div className="product-container">
                  {this.renderSections(Object.keys(item)[0], item[Object.keys(item)[0]])}
                </div>
              );
            })}
          </div>
        </div>
      </div>
    );
  }
}

ProductListGrid.propTypes = {
  product: PropTypes.shape([]),
  className: PropTypes.string,
  queryOrigin: PropTypes.string,
};

ProductListGrid.defaultProps = {
  product: [],
  className: '',
  queryOrigin: '',
};

export default withStyles(ProductListGrid, styles);
export { ProductListGrid };
