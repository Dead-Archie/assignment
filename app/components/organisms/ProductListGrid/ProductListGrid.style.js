import { css } from 'styled-components';

export default css`
  margin-top: 2rem;
  padding: 0 0 0 2rem;

  .product-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: nowrap;
    padding: 0.75rem 0px 0.375rem;
    border-bottom: 1px solid rgb(68, 68, 68);
  }

  .container-Background {
    height: 100%;
    color: rgb(255, 152, 0);
    background: rgb(51, 51, 51);
    max-width: 300px;
    box-shadow: rgba(0, 0, 0, 0.16) 0px 2px 2px 0px, rgba(0, 0, 0, 0.08) 0px 0px 0px 1px;
    border-radius: 0.625rem;
    overflow: hidden;
  }
  .image-container {
    position: relative;
    width: 300px;
    height: 300px;
  }
  .info-container {
    padding: 1rem;
  }

  .product-row {
    width: 100%;
  }
  .product-key {
    float: left;
    width: 30%;
    text-transform: uppercase;

    font-size: 0.7rem;
    font-weight: 400;
    color: rgb(158, 158, 158);
  }
  .product-value {
    color: #bab53d;
    float: right;
    width: 70%;
    text-align: right;

    font-size: 0.9rem;
    font-weight: 200;
    padding: 0px;
    margin: 0px;
  }
  .product-image {
    width: 100%;
  }
  .product-name {
    width: 100%;
    opacity: 0.8;
    position: absolute;
    bottom: 0px;
    background: rgb(32, 35, 41);
    padding: 0.625rem;
  }
  .product-title {
    font-weight: bold;
    font-size: 16px;
  }
`;
