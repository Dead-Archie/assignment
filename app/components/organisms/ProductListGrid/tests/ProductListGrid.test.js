import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { ProductListGrid } from '../ProductListGrid';

const sampleProps = {
  className: 'withStyles-sc-4wveuc-0 lgzevM',
  queryOrigin: '',
  product: {
    gender: 'male',
    name: 'Morty Smith',
    origin: {
      name: 'Earth (C-137)',
      url: 'https://rickandmortyapi.com/api/location/1',
    },
    species: 'Human',
    status: 'Alive',
    type: '',
    location: {
      name: 'Earth (Replacement Dimension)',
      url: 'https://rickandmortyapi.com/api/location/20',
    },
  },
};
const setUp = props => {
  return shallow(<ProductListGrid {...props} />);
};

describe('<ProductListGrid />', () => {
  let ProductListGridPageComponent;
  beforeEach(() => {
    ProductListGridPageComponent = setUp(sampleProps);
  });

  test('should render correctly', () => {
    expect(ProductListGridPageComponent).toMatchSnapshot();
  });
});
