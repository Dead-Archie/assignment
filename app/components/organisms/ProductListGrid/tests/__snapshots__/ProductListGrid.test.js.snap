// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductListGrid /> should render correctly 1`] = `
<div
  className="lgzevM col-6-small col-3"
>
  <div
    className="container-Background"
  >
    <div
      className="image-container"
    >
      <Image
        className="product-image"
        elementLocator=""
        inheritedStyles=""
        loading="auto"
        placeholderSrc=""
        sizes=""
        srcset=""
      />
      <div
        className="product-name"
      >
        <div
          className="product-title"
        >
          Morty Smith
        </div>
        <span>
          ID :undefined
        </span>
      </div>
    </div>
    <div
      className="info-container"
    >
      <div
        className="product-container"
      >
        <div
          className="product-row"
        >
          <div
            className="product-key"
          >
            Status
          </div>
          <div
            className="product-value"
          >
            Alive
          </div>
        </div>
      </div>
      <div
        className="product-container"
      >
        <div
          className="product-row"
        >
          <div
            className="product-key"
          >
            Species
          </div>
          <div
            className="product-value"
          >
            Human
          </div>
        </div>
      </div>
      <div
        className="product-container"
      >
        <div
          className="product-row"
        >
          <div
            className="product-key"
          >
            Gender
          </div>
          <div
            className="product-value"
          >
            male
          </div>
        </div>
      </div>
      <div
        className="product-container"
      >
        <div
          className="product-row"
        >
          <div
            className="product-key"
          >
            Origin
          </div>
          <div
            className="product-value"
          >
            Earth (C-137)
          </div>
        </div>
      </div>
      <div
        className="product-container"
      >
        <div
          className="product-row"
        >
          <div
            className="product-key"
          >
            Last Location
          </div>
          <div
            className="product-value"
          >
            Earth (Replacement Dimension)
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
